src/main/kotlin/no/fdk/dataservicecatalog/controller/DataServiceController.kt

```diff
@@ -1,48 +1,45 @@
 package no.fdk.dataservicecatalog.controller
 
 import jakarta.validation.Valid
-import no.fdk.dataservicecatalog.domain.DataServiceCount
-import no.fdk.dataservicecatalog.domain.PatchRequest
 import no.fdk.dataservicecatalog.domain.DataService
+import no.fdk.dataservicecatalog.domain.PatchRequest
 import org.springframework.http.HttpStatus
 import org.springframework.http.MediaType
 import org.springframework.http.ProblemDetail
 import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
 import org.springframework.web.bind.MethodArgumentNotValidException
 import org.springframework.web.bind.annotation.*
 
 @RestController
-@RequestMapping("/internal/catalogs")
+@RequestMapping("/internal/catalogs/{catalogId}/data-services")
 class DataServiceController {
 
-    @GetMapping("/count", produces = [MediaType.APPLICATION_JSON_VALUE])
-    fun getDataServiceCountsForPermittedCatalogs(): ResponseEntity<List<DataServiceCount>> {
-        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build()
-    }
-
-    @GetMapping("{catalogId}/data-services", produces = [MediaType.APPLICATION_JSON_VALUE])
+    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+    @PreAuthorize("hasAnyAuthority('system:root:admin', 'organization:' + #catalogId + ':admin', 'organization:' + #catalogId + ':write', 'organization:' + #catalogId + ':read')")
```

Comment:
mulig å legge disse til som globale konstanter et sted?

```
const val READ_ROLE_AUTH = "hasAnyAuthority('system:root:admin', 'organization:' + #catalogId + ':admin', 'organization:' + #catalogId + ':write', 'organization:' + #catalogId + ':read')"
```
og så:
```suggestion
    @PreAuthorize(READ_ROLE_AUTH)
```

Reply:
Skal prøve på det.